components/calendar/src/error.rs

```diff
@@ -199,7 +299,117 @@ pub enum DateError {
     NotEnoughFields,
 }
 
-impl core::error::Error for DateError {}
+impl core::error::Error for DateFromFieldsError {}
+
+impl From<RangeError> for DateFromFieldsError {
+    #[inline]
+    fn from(value: RangeError) -> Self {
+        DateFromFieldsError::Range(value)
+    }
+}
+
+/// Internal narrow error type for functions that only fail on unknown eras
+pub(crate) struct UnknownEraError;
+
+impl From<UnknownEraError> for DateError {
+    #[inline]
+    fn from(_value: UnknownEraError) -> Self {
+        DateError::UnknownEra
+    }
+}
+
+impl From<UnknownEraError> for DateFromFieldsError {
+    #[inline]
+    fn from(_value: UnknownEraError) -> Self {
+        DateFromFieldsError::UnknownEra
+    }
+}
+
+/// Internal narrow error type for functions that only fail on invalid month codes
+pub(crate) struct InvalidMonthCodeError;
```

Comment:
praise: appreciate the comment that this is internal since if this ever were to be public it should probably be an enum with a single variant